src/coreclr/gc/gc.cpp

```diff
@@ -25019,7 +25032,22 @@ void gc_heap::check_heap_count ()
 
             // the middle element is the median overhead percentage
             float median_percent_overhead = percent_overhead[1];
-            dprintf (6666, ("median overhead: %d%%", median_percent_overhead));
+
+            // apply exponential smoothing over 3 median_percent_overhead readings
+            const float smoothing_factor = 0.333f;
+            float smoothed_median_percent_overhead = dynamic_heap_count_data.smoothed_median_percent_overhead;
+            if (smoothed_median_percent_overhead != 0.0f)
+            {
+                // average it with the previous value
+                smoothed_median_percent_overhead = median_percent_overhead*smoothing_factor + smoothed_median_percent_overhead*(1.0f - smoothing_factor);
+            }
```

Comment:
instead of 0.333f, it seems a bit better to just do `median_percent_overhead / smoothing + (smoothed_median_percent_overhead / smoothing) * (smoothing - 1)` like what we do in the `exponential_smooth` method. it's consistent and more elegant. or you could refactor the `exponential_smooth` method and call it here.

Reply:
Ok, guess I was trying to avoid the floating point divisions... but once per GC it hardly matters.

Reply:
I see; makes sense. another option is to change the `exponential_smoothing` implementation, if you want to avoid division... my point was simply that there's an inconsistency.